### PR TITLE
Layouts for partials

### LINKED ISSUE #1621: Layout within PartialView is not rendered
This was working on ASP.net MVC 5

I have many partial views within a layout
Each partial view has its own layout

Previously in current version, it works flawlessly

On vnext within visual studio 2015 preview,
Setting "Layout" within partial view is not rendered at all

This is quite huge breaking change

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes #1621

### CODE REVIEW

**Comment by yishaigalatzer** [comment]
:shipit: from me, get another reviewer though to signoff

File: src/Microsoft.AspNet.Mvc.Razor/RazorView.cs [hunk 1]
```diff
@@ -62,32 +62,10 @@ private bool EnableInstrumentation
...
-            else
-            {
-                // For the non-instrumented case, we don't need to buffer contents. For Html.Partial, the writer is
-                // an in memory writer and for Partial views, we directly write to the Response.
-                await RenderPageCoreAsync(RazorPage, context);
-            }
+            // Partials don't execute _ViewStart pages, but may execute Layout pages if the Layout property
+            // is explicitly specified in the page.
+            var bodyWriter = await RenderPageAsync(RazorPage, context, executeViewStart: !IsPartial);
+            await RenderLayoutAsync(context, bodyWriter);
```
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Razor/RazorView.cs]: Do we have a test in place for this behavior for `ViewViewComponentResult`?

I remember us wanting the original behavior here both for partials and views used by view component results.

Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Razor/RazorView.cs]: Looks like you added one :heart:

File: test/Microsoft.AspNet.Mvc.FunctionalTests/ViewEngineTests.cs [hunk 2]
```diff
@@ -289,5 +305,80 @@ public static IEnumerable<object[]> RazorViewEngine_UsesExpandersForLayoutsData
...
+            // Arrange
+            var expected = "Partial that does not specify Layout";
+            var server = TestServer.Create(_provider, _app);
+            var client = server.CreateClient();
+
+            // Act
+            var body = await client.GetStringAsync("http://localhost/PartialsWithLayout/PartialDoesNotExecuteViewStarts");
+
+            // Assert
+            Assert.Equal(expected, body.Trim());
```
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/ViewEngineTests.cs]: Would also like to see some verification that `_ViewStart`s aren't rendered when a `ViewComponent` runs a view.

Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/ViewEngineTests.cs]: @pranavkm does the above make sense? How does the view engine know **not** to include view starts when rendering a partial/view-component view?

Comment (pranavkm) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/ViewEngineTests.cs]: View components are executed as partial views and consequently have the same-ish behaviors as `PartialResult` \ `RenderPartial` \ `PartialAsync`. I'll go ahead and add a `_ViewStart` to the current example to cover ViewComponents.

Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/ViewEngineTests.cs]: Care to expand on this?

> How does the view engine know not to include view starts when rendering a partial/view-component view?

Pointer to code is fine.

Comment (pranavkm) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/ViewEngineTests.cs]: It's part of the PR - https://github.com/aspnet/Mvc/pull/1846/files#diff-bf0a1f7364ebf0b4627661599633b1b4R67

**Comment by rynowak** [comment]
:shipit: after adding a test for VS + viewstart

**Pull request closed without merging**